source/fe/mapping_q_generic.cc

```diff
@@ -1614,11 +1615,16 @@ namespace internal
                 VectorizedArray<double> *values_dofs_ptr[n_comp];
                 data.values_quad.resize(n_comp*n_q_points);
                 VectorizedArray<double> *values_quad_ptr[n_comp];
+                // Some evaluators need to write into the gradients array.
+                data.gradients_quad.resize (n_comp*n_q_points*dim);
+                VectorizedArray<double> *gradients_quad_ptr[n_comp][dim];
 
                 for (unsigned int c=0; c<n_comp; ++c)
                   {
                     values_dofs_ptr[c] = &(data.values_dofs[c*n_shape_values]);
                     values_quad_ptr[c] = &(data.values_quad[c*n_q_points]);
+                    for (unsigned int j=0; j<dim; ++j)
+                      gradients_quad_ptr[c][j] = &(data.gradients_quad[(c*dim+j)*n_q_points]);
```

Comment:
When looking at this code with all that pointer setting I wonder whether we should maybe change the interface used for `FEEvaluationImpl`: We could pass a single pointer into the function and simply apply the offsets in the `gradients_quad` pointer according to this formula. The more I work with the internals of these classes, the more I am convinced that computing the offsets once or twice inside the functions is easier than setting them in any call site. That would also simplify the work for the upcoming DG kernels.